lib/tls.js

```diff
@@ -860,6 +863,11 @@ function SecurePair(credentials, isServer, requestCert, rejectUnauthorized,
     this.ssl.setNPNProtocols(options.NPNProtocols);
     this.npnProtocol = null;
   }
+  
+   // TODO:PSK 
+  if (pskCallback) {
```

Comment:
Shouldn't there be some sort of check to make sure that the ssl arguments are internally consistent? For example, would it make sense to use sni with psk? Put another way, can someone configure both PSK and normal CA based handshakes and have them work depending on what the requester asks for or does one negate the other? I see below that you disable the SNICallback if PSK is being used but in that case shouldn't a TLS request that included SNI and PSK together be refused for having an internally inconsistent options configuration?